### DISCUSSION TITLE
"Unexpected end of JSON input"

### DISCUSSION THREAD

**Discussion by mimranzulkfal** [prisma/prisma#18525] [Q&A]
### Bug description

Hi! I am using `prisma client 2.24.1`. My APIs were working before. But now they are returning this error
`Unexpected end of JSON input at JSON.parse (<anonymous>)","    at /var/task/packages/redwood/node_modules/@prisma/client/runtime/index.js:27585:31","`
whenever I call POST method. It does save the data in Supabase database but doesnt return the JSON object.

Kindly guide me why it causing this issue?.
Thanks

### How to reproduce

I just hit my Netlify graphql function, that stores the values to Supabase, but throws error after that at 
`/var/task/packages/redwood/node_modules/@prisma/client/runtime/index.js:27585:31`

### Expected behavior

It was working before and return me the JSON object.

### Prisma information

Heres my schema.prisma
```
datasource DS {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = "native"
}

// Define your own datamodels here and run `yarn redwood prisma migrate dev`
// to create migrations for them and apply to your dev DB.
model User {
  id                       String             @id
  email                    String             @unique
  name                     String?
  createdAt                DateTime           @default(now())
  updatedAt                DateTime           @default(now()) @updatedAt
  lastEntry                DateTime?
  checkInReminders         Boolean            @default(false)
  habitReminders           Boolean            @default(false)
  checkInEmailReminders    Boolean            @default(false)
  habitEmailReminders      Boolean            @default(false)
  checkInPushReminders     Boolean            @default(false)
  habitPushReminders       Boolean            @default(true)
  showGuides               Boolean            @default(true)
  showLatestEntry          Boolean            @default(true)
  showOnboarding           Boolean            @default(true)
  currentSubscriptionLevel String?
  userPractices            String[]
  userPracticeOrder        String[]
  reminders                ReminderSchedule[]
  characteristics          String?
}

model ReminderSchedule {
  id           String   @id // user.id::type::int|string (where int|string is the index/autoincrement OR id of the reminder schedule, i.e. morning = 0, afternoon = 1, in the future this could handle arbitrary schedules by allowing a user to add any number of reminders and they would be idempotently id'd by user.id::and the autoincrement id of the reminder)
  user         User     @relation(fields: [userId], references: [id])
  userId       String // relation scalar field (used in the `@relation` attribute above)
  type         String
  enabled      Boolean  @default(true)
  createdAt    DateTime @default(now())
  updatedAt    DateTime @default(now()) @updatedAt
  scheduledFor String 
}
```

### Environment & setup

- Database: PostgreSQL Supabase
- Node.js version: 16.17.1

Kindly guide me what will be the cause for this error? 
Thanks

### Prisma Version

- Prisma Client: 2.24.1

**Comment by nurul3101** [comment]
Hello @mimranzulkfal 👋 

Do you get this error on a particular prisma query?
It's difficult to predict what could be going wrong, do you have any additional database logs that could hint what could be going wrong?

**Reply by mimranzulkfal** [reply]
Yes when I try to update the user. heres the query

```
interface UpdateUserArgs extends Prisma.UserWhereUniqueInput {
  id: any,
  input: Prisma.UserUpdateInput
}

export const updateUser = ({ id, input }: UpdateUserArgs) => {
  isUserOrAdmin(id)
  return db.user.update({
    data: input,
    where: { id },
  })
}
```

db is 

```
import { PrismaClient } from '@prisma/client'

import { emitLogLevels, handlePrismaLogging } from '@redwoodjs/api/logger'

import { logger } from './logger'

/*
 * Instance of the Prisma Client
 */
export const db = new PrismaClient({
  log: emitLogLevels(['info', 'warn', 'error']),
})

handlePrismaLogging({
  db,
  logger,
  logLevels: ['info', 'warn', 'error'],
})

```

**Reply by nurul3101** [reply]
Apologies for the delay in getting back to you.

If you are still facing this issue, I would recommend adding a [DEBUG environment variable ](https://www.prisma.io/docs/concepts/components/prisma-client/debugging#setting-the-debug-environment-variable) and sharing the additional logs so we can further investigate.

**Reply by nurul3101** [reply]
It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@mimranzulkfal Do let us know in case you still need any assistance.

**Comment by nicholasmordecai** [comment]
So.. funnily enough I have the same issue. It was after I put down a project for a long time and picked it back up again after some months. 

It seems to run on every single query I try to execute. Here's a full stack trace:
```sh
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at Object.raw (/Users/nicholasmordecai/Documents/maptr/node_modules/@prisma/client/runtime/index.js:36798:18)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.request (/Users/nicholasmordecai/Documents/maptr/node_modules/@prisma/client/runtime/index.js:37468:43)
    at async cb (/Users/nicholasmordecai/Documents/maptr/node_modules/@prisma/client/runtime/index.js:38661:26) {
  clientVersion: '3.8.1'
}
```

For more context, here's my javascript:
```ts 
const data =  [
  {
    id: 'fac7cfb1-2c72-4050-acd4-eee858087efa',
    name: 'Pat Rempel',
    email: 'Lula60@hotmail.com',
    password: 'dF750P6SW9Y5aja',
    active: true,
    approved: true,
    created_at: 2023-05-18T19:23:08.134Z,
    updated_at: 2023-05-18T19:23:08.134Z
  }
];

const dataImport = await prisma.users.createMany({
  data: data,
});
```

My prisma schema for this table is:

```
model users {
  id         String   @id @default(uuid())
  name       String
  email      String   @unique
  password   String
  active     Boolean  @default(true)
  approved   Boolean  @default(false)
  tokens     tokens[]
  created_at DateTime @default(now())
  updated_at DateTime @updatedAt

  user_channel user_channel[]
  @@index([name])
  @@index([email])
}
```

**Reply by nicholasmordecai** [reply]
Also, upgrading did fix the issue, but I'm not sure why it was an issue to begin with.

For anybody else: https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4

**Reply by nurul3101** [reply]
@nicholasmordecai Thanks for sharing that upgrading the prisma version fixed the issue for you.
I am not sure what could have caused this issue as the error started happening randomly. It would be helpful if we can get a minimal reproduction for this error.

**Comment by entrptaher** [comment]
To my future self, or anyone else who encounters this. I faced this in a 4 year old project, where running that without upgrading anything is a must.

For me, it caused due to a nodejs version mismatch. My prisma 2.23 version worked properly with node 14 without this error, but when I tried to use prisma 2.23 with node 16, this error popped up from `undici2` file.

```js
const data = JSON.parse(await getStream(body2));
```

Switching to node version 14 resolved the issue for me.
